packages/cli/fern-definition/schema/src/schemas/SingleUnionTypeSchema.ts

```diff
@@ -1,19 +1,23 @@
 import { z } from "zod";
 import { SingleUnionTypeKeySchema } from "./SingleUnionTypeKeySchema";
 import { WithNameAndDocsSchema } from "./WithNameAndDocsSchema";
+import { WithAvailabilitySchema } from "./WithAvailabilitySchema";
 
 export const SingleUnionTypeSchema = z.union([
     z.string(),
-    WithNameAndDocsSchema.extend({
-        type: z.optional(
-            z.union([
-                z.string(),
-                // allow "{}" as syntactic sugar for null
-                z.record(z.never())
-            ])
-        ),
-        key: z.optional(z.union([z.string(), SingleUnionTypeKeySchema]))
-    })
+    WithAvailabilitySchema.extend(
+        WithNameAndDocsSchema.extend({
+            type: z.optional(
+                z.union([
+                    z.string(),
+                    // allow "{}" as syntactic sugar for null
+                    z.record(z.never())
+                ])
+            ),
+            key: z.optional(z.union([z.string(), SingleUnionTypeKeySchema])),
+            "display-name": z.optional(z.string())
+        }).shape
+    )
```

Comment:
nit, this would read cleaner if we did: 

```ts
z.scriptObject({
  type: z.optional(...),
  key: z.optional(...),
)}.extend([WithNameAndDocsSchema.shape, WithAvailabilitySchema.shape))
```